Marketplace/voiceHUDApp/appResources/appData/voiceHUD_app.js

```diff
@@ -0,0 +1,836 @@
+            } else {
+                audioLevel = (val - (LOUDNESS_FLOOR - 1)) * LOUDNESS_SCALE;
+            }
+            if (audioLevel > 1.0) {
+                audioLevel = 1;
+            }
+            return audioLevel;
+        }
+
+        // the VU meter should work similarly to the one in AvatarInputs: log scale, exponentially averaged
+        // But of course it gets the data at a different rate, so we tweak the averaging ratio and frequency
+        // of updating (the latter for efficiency too).
+        var audioLevel = 0.0;
+        var avgAudioLevel = 0.0;
+
+        if (user) {
+            // we will do exponential moving average by taking some the last loudness and averaging
+            user.accumulatedLevel = AVERAGING_RATIO * (user.accumulatedLevel || 0) + 
+                (1 - AVERAGING_RATIO) * (user.audioLoudness);
+
+            // add 1 to insure we don't go log() and hit -infinity.  Math.log is
+            // natural log, so to get log base 2, just divide by ln(2).
+            audioLevel = scaleAudio(Math.log(user.accumulatedLevel + 1) / LOG2);
+
+            // decay avgAudioLevel
+            avgAudioLevel = Math.max((1 - AUDIO_PEAK_DECAY) * (user.avgAudioLevel || 0), audioLevel).toFixed(3);
+
+        }
+
+        userStore[uuid].audioLevel = audioLevel;
+        userStore[uuid].avgAudioLevel = avgAudioLevel;
+    }
+
+
+    // returns an array of avatarPaldata
+    function allAvatars() {
+        return AvatarManager.getPalData().data;
+    }
+    
+
+    // searches the user list to get the index of a user
+    function getIndexOfSettingsUser(uuid) {
+        if (settings.users.length) {
+            var index = settings.users.map(function (item) {
+                return item.uuid;
+            }).indexOf(uuid);
+            return index;
+        }
+        return -1;
+    }
+
+
+    // This function gets data to sort through
+    function sortData() {
+        var avatarList = Object.keys(userStore);
+        settings.users = avatarList.map(function (uuid) { 
+            return userStore[uuid]; 
+        });
+    }
+    
+
+    // This function create a HUD element over a user's head 
+    var AVATAR_HEIGHT_OFFSET = 1.5,
+        HUD_DEFAULT_DIMENSIONS = { x: 0.05, y: 0.001, z: 0.01 },
+        HUD_MIN_Z_DIMENSIONS = 0.01,
+        HUD_MAX_Z_DIMENSIONS = [
+            0.1445,
+            0.2451,
+            0.3669,
+            0.1980,
+            0.0911
+        ],
+        HUD_ALPHA = 0.5,
+        BAR_ALPHA = 1;
+    function createHUD(uuid) {
+        var user = userStore[uuid];
+        user.hudID = [];
+        user.audioBarIDs = [];
+        user.userHeight = AvatarManager.getAvatar(uuid).getNeckPosition().y + AvatarManager.getAvatar(uuid).scale * AVATAR_HEIGHT_OFFSET;
+        var hudPosition = {x: AvatarManager.getAvatar(uuid).position.x, y: user.userHeight, z:AvatarManager.getAvatar(uuid).position.z}; // user.currentPosition
```

Comment:
Same here, what you want to do probably is 
var avatarPosition = AvatarManager.getAvatar(uuid).position;
then avatarPosition.x, .y, .z...